components/01-atoms/controls/text-link/_text-link.scss

```diff
@@ -58,12 +48,85 @@
     transform: var(--link-styles-#{$style}-transform);
   }
 
+  &:hover {
+    color: var(--link-color-hover);
+  }
+
   // Animate the decorative bar when the link is hovered
   &:hover::after {
     transform: var(--link-styles-#{$style}-transform-hover);
   }
 
-  &:focus::after {
+  &:focus-visible::after {
+    transform: scale(0, 1);
+  }
+}
+
+// This mixin is used to style linked headings in components that should be
+// "fully clickable". e.g. When an entire card is clickable, this applies the
+// appropriate styles to the heading inside the card.
+// NOTES:
+// - Apply the mixin to the parent item. e.g.: `.card` and it will affect
+// the `.card__heading-link` element inside.
+// - The component will likely also need some JS to round out the design and
+// functionality desired. Check `news-card.js` for an example. And/or read the
+// Cards chapter here: https://inclusive-components.design/cards.
+@mixin clickable-component-heading-link {
+  // This sets `$selector` to the relative class, so that later, we can style
+  // The `__heading-link` that applies to the appropriate component.
+  $selector: &;
+
+  // Style the component when it's heading is focused.
+  &:focus-within {
+    @include tokens.focus-styles;
+  }
+
+  // Style the heading link inside the component.
+  #{$selector}__heading-link {
```

Comment:
Another valid approach here may be to drop the `$selector` part of this mixin and just use `[class*='__heading-link']`. This might simplify the mixin a bit because you'll be doing less mapping. 

Either way, I think this mixin is understandable.

Reply:
I like this better. It's still scoped to the component, so there shouldn't be any spillage, and I like not having the interpolated variable in there. 👍